partiql-parser/src/main/kotlin/org/partiql/parser/internal/PartiQLParserDefault.kt

```diff
@@ -2189,6 +2214,39 @@ internal class PartiQLParserDefault : PartiQLParser {
             typeCustom(ctx.text.uppercase())
         }
 
+        override fun visitTypeComplexLegacy(ctx: GeneratedParser.TypeComplexLegacyContext) = translate(ctx) {
+            when (ctx.datatype.type) {
+                GeneratedParser.STRUCT, GeneratedParser.TUPLE -> typeStruct(emptyList())
+                GeneratedParser.ARRAY, GeneratedParser.LIST -> typeArray(null)
```

Comment:
Based on the grammar for `TypeComplexLegacy`, array won't be a possible type:

```
datatype=(STRUCT|TUPLE|LIST)                                                                                     # TypeComplexLegacy
```

Perhaps `ARRAY` would need added to the `TypeComplexLegacy` node?

Reply:
Array is not previously allowed, Array here should be a dead branch. I will remove it.

Reply:
I see what you saying is that parse of ARRAY with out any element type will not be allowed if we do not put it there. Let me change the name legacy to unparameterized.